pandas/io/parsers.py

```diff
@@ -3208,8 +3208,7 @@ def _get_empty_meta(columns, index_col, index_names, dtype=None):
         for k, v in compat.iteritems(_dtype):
             col = columns[k] if is_integer(k) else k
             dtype[col] = v
-
-    if index_col is None or index_col is False:
+    if index_col is None or index_col is False or index_names is None:
```

Comment:
Is it possible to simplify this to just be `if not (index_col or index_names)`?

Reply:
Nope. ìndex_col`can be a list ( like in the issue #21141 ) and `bool([0] or None)` is `True`

Reply:
can you add a comment here on what is going on

Reply:
If I can help, the reason the error happens is because `index_names` is `None` and it is being iterated over in https://github.com/r00ta/pandas/blob/master/pandas/io/parsers.py#L3215
To prevent this, there's an additional check for it being not None. 
@r00ta you could add a comment here saying `# also create empty index if index_names is None`
@jreback is that clear enough?

Reply:
I mean in the code itself, e.g. why we are checking these things

Reply:
@WillAyd if it was possible to simplify it, it would be `if not (index_col and index_names)` but I suspect it would be not be easy due to the complexity of the code. They can be `None` or other values like `0` and the code was checking for specific things.

Reply:
@jreback the only thing that changed here is the addition of `index_names` in the if statement so as to divert the code flow into the creation of the empty index. Hard to determine why the check is being made here for the other values. Maybe @r00ta can explain further.

Reply:
@cgopalan exactly. According to the issue #21141 if `nrow=0, `index_names`is `None` it raises an Exception. To avoid this i check that `index_names` is not `None`. If it's better to handle this corner case in another way let me know!

Reply:
@r00ta I tried other ways but they seemed to go into other code flows with complicated things further. I am not a member of the pandas team, just trying to contribute to this PR by reviewing it. AFAIK, the code is fine with me. Just need to address the comments on commenting the code and on the test.

Reply:
My hesitation is that this is already somewhat convoluted, and when that's the case typically adding things just because "they work" tends to yield some other issue down the road.

I'm not an expert on this module but can you check the following:
- There is a function called `_clean_index_names` that I assume gets called before all of this. If there's no index column it returns `None`, but what if that returned an empty list instead? 
- There is a function called `_is_index_col` in the module which appears to do the first 2/3 items of this conditional - should we be using that here?

AFAICT with the former the "cleaning" is supposed to help when the `index_cols` are `None` in instances like below, so maybe that function just shouldn't be returning `None`?

https://github.com/r00ta/pandas/blob/1d680be82e757d79fbf3e3d6d87de3586ea5e06a/pandas/io/parsers.py#L1767

https://github.com/r00ta/pandas/blob/1d680be82e757d79fbf3e3d6d87de3586ea5e06a/pandas/io/parsers.py#L2112

Reply:
@WillAyd if i change the code such that `index_names` is `[]` instead of `None` it branches to the `else` condition and gives the following error at [this line](https://github.com/r00ta/pandas/blob/1d680be82e757d79fbf3e3d6d87de3586ea5e06a/pandas/io/parsers.py#L3215)
`ValueError: Must pass non-zero number of levels/labels`

It seems like the right branch would be to go the `if index_col is None or index_col is False:` but since the `index_col` passed is an int, this is being bypassed.

The real culprit seems to be the fact that `nrows` is zero. 

Why does pandas allow `nrows=0` as a value for `read_csv`? Shouldnt it send back an error saying nrows should be greater than zero? Is there a value to reading 0 rows into a dataframe?

If we do allow `nrows` to be zero, then we can alter `index_col` to be `False` since any other value for it seems meaningless. I can look further into that, but just was curious about `nrows` allowing zero.

Reply:
I can't think of a good use case for `nrows` to be 0 so alternately could just raise if that is the case instead of all of the changes here (unless someone else does have a use case for that)

Reply:
@WillAyd The parser already checks that the parameter `nrows`is not a negative integer ( https://github.com/pandas-dev/pandas/blob/master/pandas/io/parsers.py#L382 ). It raises `ValueError: 'nrows' must be an integer >=0`. For this reason i thought that by contract the case `nrows=0` is allowable

Reply:
@WillAyd @r00ta The solution where we validate nrows to be >= 1 looks good to me. This will take care of this weird corner case.

The changes for that can be made [here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/io/test_excel.py#L998) to pass in a min_val argument of 1.
Also the tests need to be changed [here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/io/parser/common.py#L367) and [here](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/io/test_excel.py#L998)

Reply:
@cgopalan do you want to submit that as a separate PR? I think that would be preferable to this.

Just glancing at your code / tests make sure you cover the case that `nrows` equals 0